### PR TITLE
Add option to save current library grouping

### PR DESCRIPTION
**Pull request by nicklan**
- Adds saved groupings to the Group by menu
- Currently missing a way to manage saved groupings
- Saved groupings are in a new section of the config file as
  "GroupingName"->ByteArray serialization

I'm planning on writing a Saved Groupings Manager, similar to the Queue Manager, that will allow adding/removing saved groupings.  I can make it part of this request, or leave it for another one.

### CODE REVIEW

File: src/library/libraryfilterwidget.cpp [hunk 1]
```diff
@@ -114,6 +118,22 @@ LibraryFilterWidget::LibraryFilterWidget(QWidget* parent)
 
 LibraryFilterWidget::~LibraryFilterWidget() { delete ui_; }
 
+void LibraryFilterWidget::UpdateGroupByActions() {
+  if (group_by_group_) {
+    disconnect(group_by_group_,0,0,0);
```
Comment (hatstand) [inline src/library/libraryfilterwidget.cpp]: `group_by_group_->disconnect()`

File: src/library/libraryfilterwidget.cpp [hunk 2]
```diff
@@ -139,6 +159,27 @@ QActionGroup* LibraryFilterWidget::CreateGroupByActions(QObject* parent) {
...
+
+  QAction* sep1 = new QAction(parent);
+  sep1->setSeparator(true);
+  ret->addAction(sep1);
+
+  // read saved groupings
+  QSettings s;
+  s.beginGroup(LibraryModel::kSavedGroupingsSettingsGroup);
+  QStringList saved = s.childKeys();
+  for (int i = 0; i < saved.size(); ++i) {
```
Comment (hatstand) [inline src/library/libraryfilterwidget.cpp]: Use a c++11 foreach

File: src/library/libraryfilterwidget.cpp [hunk 3]
```diff
@@ -139,6 +159,27 @@ QActionGroup* LibraryFilterWidget::CreateGroupByActions(QObject* parent) {
                           LibraryModel::Grouping(LibraryModel::GroupBy_Genre,
                                                  LibraryModel::GroupBy_Artist,
                                                  LibraryModel::GroupBy_Album)));
+
+  QAction* sep1 = new QAction(parent);
```
Comment (hatstand) [inline src/library/libraryfilterwidget.cpp]: Use more descriptive names for these variables.

**Comment by hatstand** [comment]
Run `make format` too

**Comment by nicklan** [comment]
Thanks for the comments.  Latest commit resolves all the issues you mentioned.  Let me know if there's anything else.

**Pull request closed without merging**